library/Haste/Image/Image.php

```diff
@@ -202,4 +202,252 @@ public static function addWatermark($image, $watermark, $position = 'br', $targe
         // Return the path to new image
         return $strCacheName;
     }
+
+    /**
+     * Gets a gallery string based on the typical Contao template.
+     *
+     * Possible options keys:
+     *
+     *  - template (default is gallery_default)
+     *
+     * See findImages() for more option keys.
+     *
+     * @param array $options
+     *
+     * @return string
+     */
+    public static function getGallery(array $options)
+    {
+        $images = static::findImages($options);
+
+        if (isset($options['sortBy'])) {
+            static::sortImages($images, $options['sortBy'], $options);
+        }
+
+        $body = [];
+
+        foreach ($images as $k => $image) {
+            $stdClass = $image['templateData'];
+            $stdClass->class = $k % 2 === 0 ? 'odd' : 'even';
+            $body['image'][] = $stdClass;
+        }
+
+        $template = new \FrontendTemplate($options['template'] ?: 'gallery_default');
+        $template->body = $body;
+        return $template->parse();
+    }
+
+    /**
+     * Prepares the data for a typical Contao gallery fetching image data, meta data,
+     * responsive images data etc. based on options.
+     *
+     * Possible options keys:
+     *
+     *  - multiSRC (mutliple UUIDS, either as a serialized string or already as array)
+     *  - singleSRC (one UUID)
+     *  - sortBy (a key specifying the sort options. See sortImages() for options and requirements (some sorting options require more info)
+     *
+     * See prepareImage() for more option keys.
+     *
+     * @param array $options
+     *
+     * return array
+     */
+    public static function findImages(array $options)
```

Comment:
You can pass an array of UUIDs. File paths are not supported because we need the models for meta data etc.